azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java

```diff
@@ -147,6 +147,30 @@ public List<Pair<ExecutionReference, ExecutableFlow>> fetchQueuedFlows(final Sta
   public List<ExecutableFlow> fetchStaleFlowsForStatus(final Status status,
       final ImmutableMap<Status, Pair<Duration, String>> validityMap)
       throws ExecutorManagerException {
+    return fetchFlowsForStatusWithTimeValidity(status, validityMap, true);
+  }
+
+
+  public List<ExecutableFlow> fetchFreshFlowsForStatus(final Status status,
+      final ImmutableMap<Status, Pair<Duration, String>> validityMap)
+      throws ExecutorManagerException {
+    return fetchFlowsForStatusWithTimeValidity(status, validityMap, false);
+  }
+
+  /**
+   *
+   * @param status the status of flows you want
+   * @param validityMap contains the status as key, the value is a pair, defining timestamp by
+   *                    subtracting a duration from now, and what event time (submit/start/update)
+   *                    this timestamp is comparing with
+   * @param looksForStale if true return flows is stale comparing to this timestamp, else find
+   *                      flows fresher than this timestamp
+   */
+  List<ExecutableFlow> fetchFlowsForStatusWithTimeValidity(
+      final Status status,
+      final ImmutableMap<Status, Pair<Duration, String>> validityMap,
+      boolean looksForStale)
```

Comment:
maybe just stale?

Reply:
hmmm, I think if later some developer see a parameter named `stale`, they might be a little confused, so I name it this